Mathlib/Tactic/Have.lean

```diff
@@ -43,7 +43,7 @@ def haveLetCore (mvarId : MVarId) (name : Option Syntax) (bis : Array Syntax)
     let (mvar1, t, p) ← elabBinders fun es => do
       let t ← match t with
       | none => mkFreshTypeMVar
-      | some t => Tactic.elabTerm t none
+      | some t => Tactic.elabTerm.go t none
```

Comment:
This change is suboptimal, it's circumventing the new `.recover` flag in the tactic state. Maybe we can revisit it and figure out what we actually want to do here...